src/streamlink/plugins/tlctr.py

```diff
@@ -2,16 +2,22 @@
 import re
 
 from streamlink.plugin import Plugin
+from streamlink.plugin.api import validate
 from streamlink.stream import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 class TLCtr(Plugin):
-
     _url_re = re.compile(r'https?://(?:www\.)?tlctv\.com\.tr/canli-izle')
-    _hls_re = re.compile(
-        r'''["'](?P<url>https?://[^/]+/live/hls/[^"']+)["']''')
+    _token_params_re = re.compile(r'var liveUrl = ".*?tlctv(.*?)";')
+    _api_url = 'https://dygvideo.dygdigital.com/live/hls/tlctvdai{0}&json=true'
```

Comment:
Since you're trying to match the `_api_url` here except for the query parameters, you should set the `_api_url` value to the API base URL (`https://dygvideo.dygdigital.com/live/hls/tlctvdai`) without any query parameters and try to match it in the regex here instead (also make the whitespace optional). Then parse the regex match as a URL, add the `json=true` parameter and serialize it again for making the API call.

Reply:
I'm not sure if I'm misunderstanding things, but the API base URL is not quite the same as the base URL extracted from the page:
```
https://dygvideo.dygdigital.com/live/hls/tlctv # base URL from page
https://dygvideo.dygdigital.com/live/hls/tlctvdai # base for API URL
```
There could be a better way to do this though.  The only changes necessary are to substitute `tlctv` for `tlctvdai` and append the `json=true` parameter.

Perhaps it would better to extract the entire URL from the page, then: `extracted_url.replace('tlctv', 'tlctvdai')` and finally, append the extra parameter get the required API URL?

Reply:
Just to add: Strictly speaking, appending `json=true` does not appear to be necessary, I see.  Without it, the request directly returns the M3U8 file...